packages/react/src/components/Pagination/Pagination.tsx

```diff
@@ -0,0 +1,174 @@
+   * @default 'medium'
+   */
+  size?: 'small' | 'medium';
+  /** Sets how compact the component will be. If true, only 5 steps will show.
+   * @default false
+   */
+  compact?: boolean;
+  /** Hides the component's previous and next button labels
+   * @default false
+   */
+  hideLabels?: boolean;
+  /** Sets the current page */
+  currentPage: number;
+  /** Total number of pages */
+  totalPages: number;
+  /**
+   * Function to be called when the selected page changes.
+   */
+  onChange: (currentPage: number) => void;
+}
+
+// eslint-disable-next-line react/display-name
+export const Pagination = forwardRef<HTMLElement, PaginationProps>(
+  (
+    {
+      nextLabel = '',
+      previousLabel = '',
+      size = 'small',
+      compact = false,
+      hideLabels = false,
+      currentPage = 1,
+      totalPages,
+      onChange,
+    }: PaginationProps,
+    ref,
+  ) => {
+    const getSteps = () => {
+      /**
+       * Number of always visible pages at the start and end.
+       */
+      const boundaryCount = 1;
+
+      /**
+       * Number of always visible pages before and after the current page.
+       */
+      const siblingCount = compact ? 0 : 1;
+
+      const range = (start: number, end: number) =>
+        Array.from({ length: end - start + 1 }, (_, i) => start + i);
+
+      if (totalPages <= (boundaryCount + siblingCount) * 2 + 3)
+        return range(1, totalPages);
+
+      const startPages = range(1, boundaryCount);
+      const endPages = range(totalPages - boundaryCount + 1, totalPages);
+
+      const siblingsStart = Math.max(
+        Math.min(
+          currentPage - siblingCount,
+          totalPages - boundaryCount - siblingCount * 2 - 1,
+        ),
+        boundaryCount + 2,
+      );
+      const siblingsEnd = siblingsStart + siblingCount * 2;
+
+      return [
+        ...startPages,
+        siblingsStart - (startPages[startPages.length - 1] ?? 0) === 2
+          ? siblingsStart - 1
+          : 'ellipsis',
+        ...range(siblingsStart, siblingsEnd),
+        (endPages[0] ?? totalPages + 1) - siblingsEnd === 2
+          ? siblingsEnd + 1
+          : 'ellipsis',
+        ...endPages,
+      ];
+    };
+
+    return (
+      <nav ref={ref}>
+        <ul className={classes.pagination}>
+          {currentPage !== 1 ? (
+            <li className={cn(classes.chevronLeft, classes[size])}>
+              <Button
+                icon={<ChevronLeftIcon />}
+                aria-label={previousLabel}
+                onClick={() => {
+                  onChange(currentPage - 1);
+                }}
+                variant={'quiet'}
+                color={'primary'}
+                size={'small'}
+              >
+                {!hideLabels && previousLabel}
+              </Button>
+            </li>
+          ) : (
+            <div className={cn(classes.whitespace)}></div>
+          )}
+          {getSteps().map((step) => {
+            const n = Number(step);
+            return isNaN(n) ? (
```

Comment:
Why not just check for `step === 'ellipsis'`? Will make it much easier to read. We control the output of `getSteps` so it should never be anything other than `ellipsis` or number :)

Reply:
I'll fix it 😊

Reply:
I made this change 
![image](https://github.com/digdir/designsystem/assets/31471142/f72cb64a-6a14-4bb8-b3a6-28de8c6e42da)

But bug below happened when clicking next and previous multiple times, so I'll revert the change

![image](https://github.com/digdir/designsystem/assets/31471142/12f3f91c-84fa-418f-86ca-cc02f8038c36)

Reply:
Is `getSteps` outputting the correct values?

Reply:
Nvm it was the key for the li for the first ellipsis that was wrong